pkgs/development/interpreters/python/mk-python-derivation.nix

```diff
@@ -57,9 +57,12 @@ python.stdenv.mkDerivation (builtins.removeAttrs attrs ["disabled"] // {
 
   inherit pythonPath;
 
-  # patch python interpreter to write null timestamps when compiling python files
-  # this way python doesn't try to update them when we freeze timestamps in nix store
+
+  # Determinism: The interpreter is patched to write null timestamps when compiling python files.
+  # This way python doesn't try to update them when we freeze timestamps in nix store.
   DETERMINISTIC_BUILD=1;
+  # Determinism: We fix the hashes of str, bytes and datetime objects.
+  PYTHONHASHSEED = 0;
```

Comment:
This only applies to the build process, right?  I assume that the actual scripts will still run with random hash seeds; but we don't want to accidentally open up NixOS to [hash collision attacks](http://www.kb.cert.org/vuls/id/903934)